File: RabbitMQ-C++ Client/Work Queues/new_task.cpp
#include <iostream>
#include <SimpleAmqpClient/SimpleAmqpClient.h>

using namespace std;
using namespace AmqpClient;

/*
All pointers are boost's smart pointers, so if the "ptr_t" variable excapes the scope, all the memories are freed and the file descripters are closed automatically.
*/
int main(int argc, char * argv[])
{
    Channel::ptr_t channel = Channel::Create("192.168.150.128");
    //DeclareQueue(const std::string & queue_name, bool passive = false, bool durable = false, bool exclusive = true, bool auto_delete = true)
    channel->DeclareQueue("task_queue", false, true, false, false);
    char szmsg[128];
    try{
        for(int x = 0; x < 10; x++){
            sprintf(szmsg, "Hello World! [%d]", x);
            BasicMessage::ptr_t msg = BasicMessage::Create();
            msg->Body((string)szmsg);
            msg->DeliveryMode(AmqpClient::BasicMessage::delivery_mode_t::dm_persistent);            
            channel->BasicPublish("", "task_queue",  msg);
            printf(" [x] Sent %s\n", szmsg);
        }
    }
    catch(MessageReturnedException &e){
        std::cout << "Message delivery error: " << e.what() << std::endl;
    }
    return 0;
}

File: RabbitMQ-C++ Client/Work Queues/new_task.py
#!/usr/bin/env python
import pika
import sys

connection = pika.BlockingConnection(
    pika.ConnectionParameters(host='192.168.150.128'))
channel = connection.channel()

channel.queue_declare(queue='task_queue', durable=True)

for x in range(10):
    message = "Hello World! [%d]"%(x)
    channel.basic_publish(
        exchange='',
        routing_key='task_queue',
        body=message,
        properties=pika.BasicProperties(
            delivery_mode=pika.spec.PERSISTENT_DELIVERY_MODE,
        ))
    print(" [x] Sent %r" % message)
connection.close()